packages/discord.js/src/structures/AutoModerationRule.js

```diff
@@ -0,0 +1,265 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const Base = require('./Base');
+
+/**
+ * Represents an auto moderation rule.
+ * @extends {Base}
+ */
+class AutoModerationRule extends Base {
+  constructor(client, data, guild) {
+    super(client);
+
+    /**
+     * The id of this auto moderation rule.
+     * @type {Snowflake}
+     */
+    this.id = data.id;
+
+    /**
+     * The guild id of this auto moderation rule.
+     * @type {Snowflake}
+     */
+    this.guildId = data.guild_id;
```

Comment:
We have a `AutoModerationRule#guild`, can't we make this property a getter of `this.guild.id`? Or what is the reasoning behind the redundancy?

Reply:
Wouldn't this mean the getter would be nullable?

This was also done because other classes do it, though they do `guild?.id ?? data.guild_id;` instead.

Reply:
guild is not nullable here so neither would the getter for its id, unless I‘m missing something. And that’s probably the reasoning in the other classes you refer to, because there guild can be null.

Reply:
Since a guild is required for this to be created, I'll just remove `this.guildId`. It can be accessed via `this.guild.id`.